src/main/java/net/minecraftforge/client/event/sound/BackgroundMusicSelectionEvent.java

```diff
@@ -0,0 +1,94 @@
+package net.minecraftforge.client.event.sound;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.audio.BackgroundMusicSelector;
+import net.minecraft.client.audio.ISound;
+import net.minecraft.client.audio.MusicTicker;
+import net.minecraft.client.audio.SimpleSound;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * Fired whenever {@link MusicTicker} tries to get the current
+ * {@link BackgroundMusicSelector} for the current situation.
+ * 
+ * When the setter below is called, the current background music selector
+ * is replaced with the user set one. The selector will wait until the current
+ * music is finished playing followed by a random delay specified by the new
+ * selector or if {@link BackgroundMusicSelector#shouldReplaceCurrentMusic}
+ * on the new selector is true.
+ * 
+ * If the setter is never called, the background music selector will default to
+ * the client call from {@link Minecraft#getBackgroundMusicSelector}.
```

Comment:
This is a very roundabout way of explaining this. And it does not really belong in this event. This event should simply be "okay, what music should be playing right now".